postgis/docker-postgis
# Dockerfile.master.template SFCGAL / PROJ refactoring.

UPDATED 2023-Feb-11:

PROPOSED CHANGES for the `Dockerfile.master.template`

- SFCGAL building  with  cgal 5.5.x-branch  +  header only build
"By utilizing the latest commit of the CGAL 5.5.x-branch and implementing a header-only build for SFCGAL,
 one can benefit from the latest CGAL patches while avoiding compatibility issues."
related: https://github.com/postgis/docker-postgis/discussions/335

- better GDAL building ( added more type )
 
- better logging / test / etc  ( internal, undocumented )
  - `/_pgis_full_version.txt`   PostGIS_Full_Version() contains the raster, sfcgal and topology versions.
  
- refresh PROJ data - workarounds: https://trac.osgeo.org/postgis/ticket/5316   ( +1MB docker image size )

MINOR changes 

-  added a warning for all templates : " NOTE: THIS DOCKERFILE IS GENERATED VIA "make update"! PLEASE DO NOT EDIT IT DIRECTLY"

-  added (master) : `LABEL org.opencontainers.image.source="https://github.com/postgis/docker-postgis"`
    * snyk proposal:  https://snyk.io/blog/how-and-when-to-use-docker-labels-oci-container-annotations/

- github checkout upgraded to "actions/checkout@v3"  ( avoid warnings )

- etc ...

SIZE increase:
* `postgis/postgis:15-master` image size increase from 555MB -> 567MB


HOW to test :
- Build the 15-master version in your local environment. ( `VARIANT=default VERSION=15-master make build` )
- check the size of the new image ( a little bigger )
- check the contents of `/_pgis_cgal_last_commit.txt`
- check the contents of `/_pgis_full_version.txt`
- check the `PostGIS_Full_Version()`
- test with your code ...
- If you managed to test it, please write a few lines about what you tested and how you tested it, to make our work easier.
